shared-ui/components/time-remaining/TimeRemaining.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from "react"
+import PrimaryButton from "../primary-button/PrimaryButton";
+import { StyledTimeRemainingHeader, StyledButtonContainer, StyledTimeRemainingContainer, StyledTimeRemainingText, StyledTimerContainer, StyledTimeRemainingDivider, StyledTimeRemainingLabel } from "./TimeRemaining.styles"
+
+const TimeRemaining = () => {
+    const [days, setDays] = useState(0);
+    const [hours, setHours] = useState(0);
+    const [minutes, setMinutes] = useState(0);
+    const [seconds, setSeconds] = useState(0);
+
+    useEffect(() => {
+        const target = new Date("02/03/2023 23:59:59");
+
+        const interval = setInterval(() => {
+            const now = new Date();
+            const difference = target.getTime() - now.getTime();
+
+            const d = Math.floor(difference / (1000 * 60 * 60 * 24));
+            setDays(d);
+
+           
+
+
+            const h = Math.floor(
+                (difference % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)
+            );
+            setHours(h);
+
+            const m = Math.floor((difference % (1000 * 60 * 60)) / (1000 * 60));
+            setMinutes(m);
+
+            const s = Math.floor((difference % (1000 * 60)) / 1000);
+            setSeconds(s);
+
+           
+        }, 1000);
+
+        return () => clearInterval(interval);
+    }, []);
+
+
+
+    return (
+        <StyledTimeRemainingContainer>
+            <StyledTimeRemainingHeader>Time Remaining</StyledTimeRemainingHeader>
+
+
+            <StyledTimerContainer>
+
+                <div>
+                    <StyledTimeRemainingText>{days}  :</StyledTimeRemainingText>
+                    <StyledTimeRemainingLabel>Days</StyledTimeRemainingLabel>
+                </div>
+
+
+                <div>
+                    <StyledTimeRemainingText>{hours}  :</StyledTimeRemainingText>
+                    <StyledTimeRemainingLabel>Hours</StyledTimeRemainingLabel>
+                </div>
+
+
+                <div>
+
+                    <StyledTimeRemainingText>{minutes}  :</StyledTimeRemainingText>
+                    <StyledTimeRemainingLabel>Minutes</StyledTimeRemainingLabel>
+                </div>
+
+
+
+                <div>
+
+                    <StyledTimeRemainingText>{seconds}</StyledTimeRemainingText>
+                    <StyledTimeRemainingLabel>Seconds</StyledTimeRemainingLabel>
+                </div>
```

Comment:
These are all pretty much the same thing except for the unit of time that's different. To reduce this code, let's create an array of TimeUnit which would look something like {text: seconds, label: 'Seconds'}. And then we can map over your array instead of having 

```
 <StyledTimeRemainingText>{minutes}  :</StyledTimeRemainingText>
                    <StyledTimeRemainingLabel>Minutes</StyledTimeRemainingLabel>
                </div>
```

this chunk 4 times